src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -864,86 +864,120 @@ private static unsafe void AdvSimdDecode(ref byte* srcBytes, ref byte* destBytes
-
-            Vector128<byte> decOne1;
-            Vector128<byte> decOne2;
-            Vector128<byte> decOne3;
-            Vector128<byte> decOne4;
-            Vector128<byte> decTwo1;
-            Vector128<byte> decTwo2;
-            Vector128<byte> decTwo3;
-            Vector128<byte> decTwo4;
-            Vector128<byte> str1;
-            Vector128<byte> str2;
-            Vector128<byte> str3;
-            Vector128<byte> str4;
-            Vector128<byte> res1;
-            Vector128<byte> res2;
-            Vector128<byte> res3;
+            // In the decoding process, we want to map each byte, representing a Base64 value, to its 6-bit (0-63) representation.
+            // It uses the following mapping. Values outside the following groups are invalid and, we abort decoding when encounter one.
+            //
+            // #    From       To         Char
+            // 1    [43]       [62]       +
+            // 2    [47]       [63]       /
+            // 3    [48..57]   [52..61]   0..9
+            // 4    [65..90]   [0..25]    A..Z
+            // 5    [97..122]  [26..51]   a..z
+            //
+            // To map an input value to its Base64 representation, we use look-up tables 'decLutOne' and 'decLutTwo'.
+            // 'decLutOne' helps to map groups 1, 2 and 3 while 'decLutTwo' maps groups 4 and 5 in the above list.
+            // After mapping, each value falls between 0-63. Consequently, the last six bits of each byte now hold a valid value.
+            // We then compress four such bytes (with valid 4 * 6 = 24 bits) to three UTF8 bytes (3 * 8 = 24 bits).
+            // For faster decoding, we use SIMD operations that allow the processing of multiple bytes together.
+            // However, the compress operation on adjacent values of a vector could be slower. Thus, we de-interleave while reading the input bytes that store adjacent
+            // bytes in separate vectors. This later simplifies the compress step with the help of logical operations.
+            // This requires interleaving while storing the decoded result.
+
+            // Values in 'decLutOne'
+            //   255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255
+            //   255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255
+            //   255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,  62, 255, 255,  63
+            //    0,  255,   0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13
+            var decLutOne = (Vector128<byte>.AllBitsSet,
+                             Vector128<byte>.AllBitsSet,
+                             Vector128.Create(0xFFFFFFFF, 0xFFFFFFFF, 0x3EFFFFFF, 0x3FFFFFFF).AsByte(),
+                             Vector128.Create(0x37363534, 0x3B3A3938, 0xFFFF3D3C, 0xFFFFFFFF).AsByte());
+
+            // Values in 'decLutTwo'
+            //   14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25, 255, 255, 255, 255
```

Comment:
likewise here. This comment needs to update to capture right values.

Reply:
Done.